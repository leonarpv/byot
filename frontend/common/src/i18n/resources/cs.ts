/**
 * Plural keys
 * 0 - singular
 * 1 - 2,3,4
 * 2 - 0 and >5
 */
export default {
  'Login': 'Přihlášení',
  'Username or email': 'Použivatelské jméno nebo email',
  'Enter username or email': 'Zadejte použivatelské jméno nebo email',
  'Enter password': 'Zadejte heslo',
  'Password': 'Heslo',
  'Forgot password?': 'Zapomenuté heslo',
  "Don't have an account?": 'Vytvořit konto',
  'Sign up': 'Registrace',
  'Enter valid email': 'Zadejte platný email',
  'Enter email': 'Zadejte email',
  'Password must be longer than 8 characters': 'Heslo musí mít alespoň 8 znaků',
  'Password must contain at least one special character': 'Heslo musí obsahovat alespoň jeden speciální znak',
  'Confirm password': 'Potvrzení hesla',
  'Passwords are not the same': 'Heslá se neshodují',
  'You have to agree with our terms and conditions': 'Musíte souhlasit se všeobecnými podmínkami',
  'First name': 'Jméno',
  'Last name': 'Přimění',
  'Email': 'Email',
  'I allow this website to collect and store submitted data.': 'Souhlasím se spracovaním osobních údajů.',
  'Already have an account?': 'Již máte konto?',
  'Log in': 'Přihlaste se',
  'Request reset password': 'Požádat o změnu hesla',
  'Reset password': 'Zabudnuté heslo',
  'All rights reserved to': 'Všechna práva vyhrazena pro',
  'Back': 'Zpět',
  'Error while trying to login, please, check username or password':
    'Nastala chyba při přihlašovaní, skontrolujte, prosím, zadané údaje',
  'Log out': 'Odhlásit se',
  'training_0': 'trénink',
  'training_1': 'tréninky',
  'training_2': 'tréninků',
  'video_0': 'video',
  'video_1': 'videa',
  'video_2': 'videí',
  'Enter training name': 'Název tréninku',
  'Add video': 'Přidat video',
  'Save training': 'Uložit trénink',
  'Video label': 'Název videa',
  'Start typing to search or paste URL': 'Vyhledejte nebo vložte URL',
  'My training sets': 'Moje seznamy tréninkov',
  'Nothing found': 'Nic nenalezeno',
  'Type to search': 'Pro vyhledávaní začněte psát',
  'Search': 'Vyhledávaní',
  'Home': 'Domů',
  'Create training set': 'Vytvořit seznam tréninků',
  'My profile': 'Můj profil',
};
