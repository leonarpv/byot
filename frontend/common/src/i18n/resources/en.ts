export default {
  'Login': 'Login',
  'Username or email': 'Username or email',
  'Enter username or email': 'Enter username or email',
  'Enter password': 'Enter password',
  'Password': 'Password',
  'Forgot password?': 'Forgot password?',
  "Don't have an account?": "Don't have an account?",
};
