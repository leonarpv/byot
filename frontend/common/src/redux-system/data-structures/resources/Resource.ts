export enum ResourceState {
  IDLE = 'IDLE',
  LOADING = 'LOADING',
}

export interface Resource<T> {
  data?: T | undefined;
  isProcessing: boolean;
  isLoaded: boolean;
  hasData: boolean;
  state: ResourceState;
}
