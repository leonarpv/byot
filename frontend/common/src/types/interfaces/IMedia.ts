export {Media as IMedia} from '../../shared/graphql/ts/types';
