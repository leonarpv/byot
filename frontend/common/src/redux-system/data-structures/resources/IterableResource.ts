import {EntityResource} from './EntityResource';
import {Resource, ResourceState} from './Resource';
import {IList} from '../../../types/interfaces/IList';

export class IterableResource<T> extends EntityResource<T[]> {
  public totalCount: number;
  public appendNewData: boolean;

  constructor(
    defaultData: T[] = [],
    {
      state = ResourceState.IDLE,
      totalCount = defaultData.length,
      appendNewData = true,
    }: {
      state?: ResourceState;
      totalCount?: number;
      appendNewData?: boolean;
    } = {}
  ) {
    super(defaultData, state);
    this.totalCount = totalCount;
    this.appendNewData = appendNewData;
  }

  get hasData(): boolean {
    return this.data!.length > 0;
  }

  setData<D = T[] | IList<any>>(value: D) {
    const data = Array.isArray(value) ? value || [] : ((value as unknown) as IList<any>).entries;
    this._data = this.appendNewData ? [...(this._data || []), ...data] : data;
    this.totalCount = Array.isArray(value)
      ? value.length
      : ((value as unknown) as IList<any>).meta?.totalCount;
  }

  get data(): T[] {
    return this._data || [];
  }

  public reset<R extends Resource<T[]> = IterableResource<T>>(): R {
    return (new IterableResource<T>(this.defaultData, {
      appendNewData: this.appendNewData,
      totalCount: this.defaultData?.length,
    }) as unknown) as R;
  }
}
