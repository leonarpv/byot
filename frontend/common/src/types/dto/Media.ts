import {IMedia} from '../interfaces/IMedia';
import {ISource} from '../interfaces/ISource';
import {IDateTime} from '../interfaces/IDateTime';
import {DateTime} from './DateTime';
import {Source} from './Source';

export class Media implements IMedia {
  public id: string;
  public label: string;
  public source: ISource;
  public updatedAt: IDateTime;
  public createdAt?: IDateTime;

  constructor({id, label, source, updatedAt, createdAt}: IMedia) {
    this.createdAt = createdAt instanceof DateTime ? createdAt : new DateTime(createdAt);
    this.updatedAt = updatedAt && updatedAt instanceof DateTime ? updatedAt : new DateTime(updatedAt);
    this.id = id;
    this.label = label;
    this.source = source instanceof Source ? source : new Source(source);
  }
}
