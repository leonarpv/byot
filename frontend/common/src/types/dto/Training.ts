import {ITraining} from '../interfaces/ITraining';
import {IList} from '../interfaces/IList';
import {IUser} from '../interfaces/IUser';
import {IDateTime} from '../interfaces/IDateTime';
import {ITrainingSet} from '../interfaces/ITrainingSet';
import {IMedia} from '../interfaces/IMedia';
import {DateTime} from './DateTime';
import {User} from './User';
import {List} from './List';

export class Training implements ITraining {
  public id: string;
  public label: string;
  public media: IList<IMedia>;
  public owner: IUser;
  public trainingSet: ITrainingSet;
  public updatedAt?: IDateTime;
  public createdAt: IDateTime;
  constructor({id, label, media, owner, trainingSet, updatedAt, createdAt}: ITraining) {
    this.createdAt = createdAt instanceof DateTime ? createdAt : new DateTime(createdAt);
    this.updatedAt = updatedAt && updatedAt instanceof DateTime ? updatedAt : new DateTime(updatedAt);
    this.id = id;
    this.label = label;
    this.media = media instanceof List ? media : new List(media);
    this.owner = owner instanceof User ? owner : new User(owner);
    this.trainingSet = trainingSet;
  }
}
