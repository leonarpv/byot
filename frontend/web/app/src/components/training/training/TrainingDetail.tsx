import * as React from 'react';
import {Grid, makeStyles, Theme} from '@material-ui/core';
import {WithStyles} from '@byot-frontend/web-common/src/types/WithStyles';
import {IterableResource} from '@byot-frontend/common/src/redux-system/data-structures/resources/IterableResource';
import {IMedia} from '@byot-frontend/common/src/types/interfaces/IMedia';
import {MediaList} from '../../media/list/MediaList';
import {MediaPlayer} from '../../player/MediaPlayer';
import {MediaPlayerSkeleton} from '../../player/MediaPlayerSkeleton';
import {TripleComboItemSkeletonList} from '../../list/TripleComboItemSkeletonList';

interface Props extends WithStyles<typeof styles> {
  media: IterableResource<IMedia>;
  onMediaClick: (media: IMedia) => void;
  currentMedia: IMedia;
}

const styles = (theme: Theme) => ({
  root: {},
  player: {
    maxHeight: '75vh',
    order: 1,
    [theme.breakpoints.up('sm')]: {
      order: 2,
    },
  },
  list: {
    order: 2,
    [theme.breakpoints.up('sm')]: {
      overflow: 'auto',
      maxHeight: '100vh',
      order: 1,
    },
  },
});
const useStyles = makeStyles(styles);

export const TrainingDetail: React.FC<Props> = (props: Props) => {
  const styles = useStyles(props);
  return (
    <Grid container spacing={2} classes={{root: styles.root}}>
      <Grid item xs={12} sm={4} md={3} classes={{root: styles.list}}>
        {props.media.isProcessing ? (
          <TripleComboItemSkeletonList />
        ) : (
          <MediaList onItemClick={props.onMediaClick} items={props.media.data} />
        )}
      </Grid>
      <Grid item xs={12} sm={8} md={9} classes={{root: styles.player}}>
        {props.media.isProcessing ? <MediaPlayerSkeleton /> : <MediaPlayer media={props.currentMedia} />}
      </Grid>
    </Grid>
  );
};
