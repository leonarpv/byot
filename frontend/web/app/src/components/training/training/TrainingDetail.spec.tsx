import * as React from 'react';
import {render, fireEvent} from '@testing-library/react';
import {TrainingDetail} from './TrainingDetail';
import {media} from '@byot-frontend/common/test/fixtures/dto/Media';
import {IMedia} from '@byot-frontend/common/src/types/interfaces/IMedia';
import {IterableResource} from '@byot-frontend/common/src/redux-system/data-structures/resources/IterableResource';
import {ResourceState} from '@byot-frontend/common/src/redux-system/data-structures/resources/Resource';
import {training} from '@byot-frontend/common/test/fixtures/dto/Training';

jest.mock('../../player/MediaPlayer', () => ({
  __esModule: true,
  MediaPlayer: () => <div data-testid="training-detail-media-player" />,
}));
jest.mock('../../player/MediaPlayerSkeleton', () => ({
  __esModule: true,
  MediaPlayerSkeleton: () => <div data-testid="training-detail-media-skeleton" />,
}));
jest.mock('../../list/TripleComboItemSkeletonList', () => ({
  __esModule: true,
  TripleComboItemSkeletonList: () => <div data-testid="training-detail-tci-skeletons" />,
}));

describe('<TrainingDetail/>', () => {
  it('should render', () => {
    const {container} = render(
      <TrainingDetail
        onMediaClick={jest.fn()}
        onDelete={jest.fn()}
        currentMedia={media()}
        media={[]}
        training={training()}
      />
    );
    expect(container).toMatchSnapshot();
  });
  it('should display skeletons', () => {
    const m = [media(), media()];
    const {container, queryByTestId, queryAllByTestId} = render(
      <TrainingDetail
        isLoading
        onMediaClick={jest.fn()}
        currentMedia={m[0]}
        media={m}
        onDelete={jest.fn()}
        training={training()}
      />
    );
    expect(queryByTestId('training-detail-media-skeleton')).not.toBeNull();
    expect(queryByTestId('training-detail-tci-skeletons')).not.toBeNull();
    expect(queryByTestId('training-detail-media-player')).toBeNull();
    expect(queryAllByTestId('media-list-item')).toHaveLength(0);
    expect(container).toMatchSnapshot();
  });
  it('should select a media', () => {
    const m = [media(), media()];
    const onMediaClick = jest.fn();
    const {container, queryByTestId, queryAllByTestId} = render(
      <TrainingDetail
        onMediaClick={onMediaClick}
        onDelete={jest.fn()}
        currentMedia={m[0]}
        media={m}
        training={training()}
      />
    );
    const listItems = queryAllByTestId('media-list-item');
    expect(queryByTestId('training-detail-media-skeleton')).toBeNull();
    expect(queryByTestId('training-detail-tci-skeletons')).toBeNull();
    expect(queryByTestId('training-detail-media-player')).not.toBeNull();
    expect(listItems).toHaveLength(2);
    fireEvent.click(listItems[1]);
    expect(onMediaClick).toBeCalledWith(m[1]);
    expect(container).toMatchSnapshot();
  });
  it('should delete media', () => {
    const m = [media(), media()];
    const onDelete = jest.fn();
    const {container, getByTestId} = render(
      <TrainingDetail
        onDelete={onDelete}
        onMediaClick={jest.fn()}
        currentMedia={m[0]}
        media={m}
        training={training()}
      />
    );
    fireEvent.click(getByTestId('app-elements-controls-remove'));
    fireEvent.click(getByTestId('app-elements-controls-confirm-yes'));
    expect(onDelete).toBeCalled();
    expect(container).toMatchSnapshot();
  });
});
