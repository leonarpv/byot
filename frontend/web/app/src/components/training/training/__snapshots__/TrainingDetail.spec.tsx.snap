// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainingDetail/> should display skeletons 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-7 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root makeStyles-list-9 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3"
    >
      <div
        data-testid="training-detail-tci-skeletons"
      />
    </div>
    <div
      class="MuiGrid-root makeStyles-player-8 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-9"
    >
      <div
        data-testid="training-detail-media-skeleton"
      />
    </div>
  </div>
</div>
`;

exports[`<TrainingDetail/> should render 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root makeStyles-list-3 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      />
    </div>
    <div
      class="MuiGrid-root makeStyles-player-2 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-9"
    >
      <div
        data-testid="training-detail-media-player"
      />
    </div>
  </div>
</div>
`;

exports[`<TrainingDetail/> should select a media 1`] = `
<div>
  <div
    class="MuiGrid-root makeStyles-root-10 MuiGrid-container MuiGrid-spacing-xs-2"
  >
    <div
      class="MuiGrid-root makeStyles-list-12 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            aria-disabled="false"
            class="MuiBox-root MuiBox-root-25 MuiButtonBase-root MuiListItem-root makeStyles-root-16 makeStyles-readonlyItem-14 MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart makeStyles-root-24"
            data-testid="media-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="makeStyles-image-18 makeStyles-image-21"
            />
            <div
              class="MuiListItemText-root makeStyles-textRoot-20 makeStyles-textRoot-23"
            >
              <div
                class="MuiTypography-root MuiListItemText-primary makeStyles-primaryText-19 makeStyles-primaryText-22 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Hu ibezvu liuz hirneom luze zakef ac aconki kekefjud voswulun kepdub optiowa ti velfo.
              </div>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
        >
          <div
            aria-disabled="false"
            class="MuiBox-root MuiBox-root-29 MuiButtonBase-root MuiListItem-root makeStyles-root-16 makeStyles-readonlyItem-14 MuiListItem-gutters MuiListItem-button MuiListItem-alignItemsFlexStart makeStyles-root-24"
            data-testid="media-list-item"
            role="button"
            tabindex="0"
          >
            <div
              class="makeStyles-image-18 makeStyles-image-26"
            />
            <div
              class="MuiListItemText-root makeStyles-textRoot-20 makeStyles-textRoot-28"
            >
              <div
                class="MuiTypography-root MuiListItemText-primary makeStyles-primaryText-19 makeStyles-primaryText-27 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Pe jilre itefihpaz zij di mu tu juf ugibo etauznop ofo fuz.
              </div>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root makeStyles-player-11 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-8 MuiGrid-grid-md-9"
    >
      <div
        data-testid="training-detail-media-player"
      />
    </div>
  </div>
</div>
`;
